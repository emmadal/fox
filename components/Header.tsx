import { View, Text, TouchableOpacity, useColorScheme } from "react-native";
import Icon from "@expo/vector-icons/Octicons";
import i18n from "@/i18n";
import { Colors } from "@/constants/Colors";
import { useNavigation, DrawerActions } from "@react-navigation/native";
import { Badge } from "./Badge";
import { notifs } from "@/data/notif";

export function Header() {
  const navigation = useNavigation();
  const colorScheme = useColorScheme();
  return (
    <View
      className="h-28 justify-end px-5 bg-white dark:bg-black"
      testID="header"
    >
      <View className="flex-row items-center justify-between mb-5">
        <View
          className="bg-transparent flex-row items-center dark:bg-transparent gap-2"
          testID="header-content"
        >
          <Icon name="flame" size={24} color="#f97316" testID="icon-flame" />
          <Text
            className="text-black font-bold text-3xl dark:text-white"
            testID="app-name"
          >
            {i18n.t("CFBundleDisplayName")}
          </Text>
        </View>

        <View className="flex-row items-center gap-14">
          <TouchableOpacity
            onPress={() =>
              navigation.navigate("(home)", { screen: "notification" })
            }
            className="bg-transparent"
            testID="notification-button"
          >
            {notifs.length === 0 ? null : (
              <Badge
                text={notifs.length > 10 ? "10+" : `${notifs.length}`}
                color="danger"
              />
            )}

            <Icon
              name="bell"
              size={25}
              color={
                colorScheme === "light" ? Colors.light.text : Colors.dark.text
              }
              testID="icon-bell"
            />
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => navigation.dispatch(DrawerActions.openDrawer())}
            className="bg-transparent"
            testID="drawer-button"
          >
            <Icon
              name="three-bars"
              size={25}
              color={
                colorScheme === "light" ? Colors.light.text : Colors.dark.text
              }
              testID="icon-gear"
            />
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}
