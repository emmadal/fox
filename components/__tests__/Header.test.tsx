import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { Header } from "../Header";
import { DrawerActions } from "@react-navigation/native";

// Mock Navigation
jest.mock("@react-navigation/native", () => {
  const actualNavigation = jest.requireActual("@react-navigation/native");
  return {
    ...actualNavigation,
    useNavigation: jest.fn(() => ({
      dispatch: jest.fn(),
    })),
    DrawerActions: {
      openDrawer: jest.fn(),
    },
  };
});

jest.mock("@expo/vector-icons/Octicons", () => {
  return {
    __esModule: true,
    default: ({ testID }: { testID: string }) => {
      const { View } = require("react-native");
      return <View testID={testID} />;
    },
  };
});

describe("Header Component", () => {
  it("renders correctly", () => {
    const { toJSON } = render(
      <NavigationContainer>
        <Header />
      </NavigationContainer>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("displays the flame icon, app name, and three-bars icon", () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <Header />
      </NavigationContainer>,
    );

    expect(getByTestId("icon-flame")).toBeTruthy();
    expect(getByTestId("app-name")).toBeTruthy();
    expect(getByTestId("icon-gear")).toBeTruthy();
  });

  it("dispatches the drawer action when the button is pressed", () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <Header />
      </NavigationContainer>,
    );

    fireEvent.press(getByTestId("header-button"));
    expect(DrawerActions.openDrawer).toHaveBeenCalled();
  });
});
