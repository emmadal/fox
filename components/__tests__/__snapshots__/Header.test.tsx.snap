// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component drawer button renders correctly 1`] = `
<View
  className="h-28 justify-end px-5 bg-white dark:bg-black"
  testID="header"
>
  <View
    className="flex-row items-center justify-between mb-5"
  >
    <View
      className="bg-transparent flex-row items-center dark:bg-transparent gap-2"
      testID="header-content"
    >
      <View
        testID="icon-flame"
      />
      <Text
        className="text-black font-bold text-3xl dark:text-white"
        testID="app-name"
      >
        Fox
      </Text>
    </View>
    <View
      className="flex-row items-center gap-14"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="notification-button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderRadius": 50,
                "height": 22,
                "justifyContent": "center",
                "position": "absolute",
                "right": -7,
                "top": -10,
                "width": 22,
                "zIndex": 1,
              },
              {
                "backgroundColor": "#f44336",
              },
            ]
          }
          testID="badge"
        >
          <Text
            className="text-center text-xs text-white"
          >
            4
          </Text>
        </View>
        <View
          testID="icon-bell"
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="drawer-button"
      >
        <View
          testID="icon-gear"
        />
      </View>
    </View>
  </View>
</View>
`;
